solang-parser/src/diagnostics.rs

```diff
@@ -1,7 +1,7 @@
 use crate::pt;
 use crate::pt::Loc;
 
-#[derive(Debug, Eq, Hash, PartialOrd, Ord, PartialEq)]
+#[derive(Clone, Debug, Eq, Hash, PartialOrd, Ord, PartialEq)]
```

Comment:
This enum does not have any data saved in it, so you can derive `Copy` instead of clone. It will just copy a `u8`.

Reply:
If I add `Copy` without `Clone` the compiler says I need to derive `Clone` as well.

I don't think you can be sure this will be represented as a `u8`. In fact, I suspect it will be a `u32` by default - that is what C does at least.